### DISCUSSION TITLE
When starting a beatmap with a non-existent leaderboard selected, it should default to Local leaderboard instead

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#23713] [Ideas]
Currently, when the Country, Global, or Friend leaderboard is unavailable either because the beatmap is unranked or because I don't have an osu!supporter tag, and I select one of those empty leaderboards and then start the beatmap, then ingame I just don't see any scores in the top left to compete against.

I think upon starting a beatmap it should switch to local scores instead if an unavailable leaderboard is selected, and when I'm done and come back to the menu then the local leaderboard should be selected as well.

Local always works as a default leaderboard.

As for not wanting to see your existing scores either while playing (maybe the animation is distracting when you beat one of your old scores), you can just leave Details selected. That should stay selected I think.

**Comment by peppy** [comment]
Sound like a decent idea, but the UX needs to be done in a way that having the game switch back to the global leaderboard when it *is* available should not feel jarring.

**Reply by ghost** [reply]
So I think I sort of implemented this (it's really simple) but I don't quite understand this requirement. Why and when should the game switch to the global leaderboard? I'm using `config.SetValue(OsuSetting.BeatmapDetailTab, PlayBeatmapDetailArea.TabType.Local)` by the way so it affects the global beatmap detail tab. If the selected leaderboard wasn't available on beatmap start, it switches to local and stays local after the user returned to song select.

It switching to global *if* the global leaderboard becomes available (if that's what you mean) seems confusing.
